backend/src/main/java/eu/clarin/switchboard/resources/ToolsResource.java

```diff
@@ -36,14 +34,40 @@ public ToolsResource(ToolRegistry toolRegistry, ToolConfig toolConfig) {
     @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
     public Response getTools(@QueryParam("mediatype") String mediatype,
                              @QueryParam("language") String language,
-                             @QueryParam("onlyProductionTools") String onlyProductionTools) {
+                             @QueryParam("onlyProductionTools") String onlyProductionTools,
+                             @Context UriInfo uriInfo) {
+        ProfileMatcher.Builder profileMatcher = ProfileMatcher.builder();
+
+        MultivaluedMap<String, String> queryParams = uriInfo.getQueryParameters();
+        for (String param: queryParams.keySet()) {
+            if ("mediatype".equalsIgnoreCase(param) ||
+                    "language".equalsIgnoreCase(param) ||
+                    "onlyProductionTools".equalsIgnoreCase(param)) {
+                continue;
```

Comment:
I'm not a huge fan of the 'goto' like 'continue' construct. Refactoring this to an if/then/else block would make it more verbose and make the code easier to understand.